packages/next/server/web/spec-extension/response.ts

```diff
@@ -44,23 +44,40 @@ export class NextResponse extends Response {
     })
   }
 
-  static redirect(url: string | NextURL | URL, status = 307) {
+  static redirect(
+    url: string | NextURL | URL,
+    status = 307,
+    init: ResponseInit = {}
+  ) {
```

Comment:
```suggestion
  static redirect(
    url: string | NextURL | URL,
    init: 307 | 308 | 301 | 302 | ResponseInit = {}
  );
```

What about retaining 2 parameters? 3 parameters add some unnecessary complexity.